### PR TITLE
refactor(meta): generate SST id without meta store write.

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Move RowIdGenerator to common crate and rename to IdGenerator.
- Reuse IdGenerator to generate SST id. The timestamp in id will be used in later SST full GC.
- Rename *table_id to *sst_id.
- Remove useless metrics.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
https://github.com/singularity-data/risingwave/issues/4037

### CODE REVIEW

File: src/common/src/util/row_id.rs [hunk 1]
```diff
@@ -19,33 +19,33 @@ const WORKER_ID_SHIFT_BITS: u8 = 12;
 const SEQUENCE_UPPER_BOUND: u16 = 1 << 12;
 const WORKER_ID_UPPER_BOUND: u32 = 1 << 10;
 
-/// `RowIdGenerator` generates unique row ids using snowflake algorithm as following format:
```
Comment (yezizp2012) [inline src/common/src/util/row_id.rs]: Please change the file name as well.

File: src/meta/src/hummock/hummock_manager.rs [hunk 2]
```diff
@@ -262,6 +265,9 @@ where
             compaction_scheduler: parking_lot::RwLock::new(None),
             compactor_manager,
             max_committed_epoch: AtomicU64::new(0),
+            // TODO: assign worker id for each meta node in meta node cluster, so that generated id
+            // is unique within the cluster.
+            sst_id_generator: Mutex::new(IdGenerator::with_epoch(0, *UNIX_SINGULARITY_DATE_EPOCH)),
```
Comment (yezizp2012) [inline src/meta/src/hummock/hummock_manager.rs:270]: Can we just generate id in client side with different worker id, so we don't need a extra RPC call.

Comment (zwang28) [reply src/meta/src/hummock/hummock_manager.rs:270]: Then each client should be assigned a worker id for its generator, which is different from the cluster node id?

File: src/common/src/util/row_id.rs (hunk 1 above)
Comment (zwang28) [reply src/common/src/util/row_id.rs]: fixed

**Comment by zwang28** [comment]
We've decided to not reply on timestamp in SST id.

**Pull request closed without merging**